### DISCUSSION TITLE
How to get localstorage value inside middleware?

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#17871] [Questions] [answered]
How to get localstorage value inside middleware?

**Comment by danielroe** [comment, marked as answer]
https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage.

But of course you will need to confirm that it's running on client-side by checking `if (process.client) {}`.

**Reply by mihail727** [reply]
> https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage.
> 
> But of course you will need to confirm that it's running on client-side by checking `if (process.client) {}`.

Can I use localstorage in middleware ?
`@/middleware/auth.ts`
```ts
export default defineNuxtRouteMiddleware(async (to, from) => {
	console.log(localstorage.getItem('token'));
});
```

**Reply by danielroe** [reply]
It's not `localstorage` but `localStorage`. And yes, you can access it anywhere in your code.... but note, as I said:

> But of course you will need to confirm that it's running on client-side by checking `if (process.client) {}`.

**Reply by mihail727** [reply]
> It's not `localstorage` but `localStorage`. And yes, you can access it anywhere in your code.... but note, as I said:
> 
> > But of course you will need to confirm that it's running on client-side by checking `if (process.client) {}`.

wow thanks
my final solution is:
```ts
export default defineNuxtRouteMiddleware(async (to, from) => {
	if (process.client) {
		console.log(localStorage.getItem('token'));
	}
});
```

**Reply by mihail727** [reply]
> It's not `localstorage` but `localStorage`. And yes, you can access it anywhere in your code.... but note, as I said:
> 
> > But of course you will need to confirm that it's running on client-side by checking `if (process.client) {}`.

one more question please ==== can i get localStorage on the server side?

**Reply by danielroe** [reply]
No, it's a browser API. You could use a cookie or something like https://github.com/nuxt-community/universal-storage-module.
